app/filevalueresolver.go

```diff
@@ -0,0 +1,69 @@
+package app
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/TIBCOSoftware/flogo-lib/config"
+	"github.com/TIBCOSoftware/flogo-lib/logger"
+)
+
+var preload = make(map[string]interface{})
+
+var log = logger.GetLogger("file-props-resolver")
+
+// Comma separated list of json files overriding default application property values
+// e.g. FLOGO_APP_PROPS_FILE_OVERRIDE=app1.json,common.json
+const EnvAppPropertyFileConfigKey = "FLOGO_APP_PROPS_FILE_OVERRIDE"
+
+func init() {
+
+	filePaths := getExternalFiles()
+	if filePaths != "" {
+		// Register value resolver
+		RegisterPropertyValueResolver("file", &FileValueResolver{})
+
+		if config.GetAppPropertiesValueResolver() == "" {
+			//Make file resolver default since FLOGO_APP_PROPS_FILE_OVERRIDE is set
+			os.Setenv(config.ENV_APP_PROPERTY_RESOLVER_KEY, "file")
+		}
+
+		// preload props from files
+		files := strings.Split(filePaths, ",")
+		if len(files) > 0 {
+			for _, filePath := range files {
+				props := make(map[string]interface{})
+
+				file, e := ioutil.ReadFile(filePath)
+				if e != nil {
+					log.Errorf("Can not read - %s due to error - %v", filePath, e)
+					panic("")
```

Comment:
@mellistibco : Most log based tooling look for **ERROR** phrase for alerting. Hence, I opt to log error and just panic. In this case, panic is needed because user intend to use this mode through env var but did not configure it properly.